openid_connect.js

```diff
@@ -7,7 +7,38 @@ var newSession = false; // Used by oidcAuth() and validateIdToken()
 
 export default {auth, codeExchange, validateIdToken, logout};
 
+function sleep(ms) {
+    return new Promise(resolve => setTimeout(resolve, ms));
+}
+
+async function isVarDefined(variable) {
+    const v = await sleep(10);
+    if (variable != null) return true;
+    else return false;
+}
+
+// If the ID token has not been synced yet, poll the variable every 10ms until
+// get a value or after a timeout.
+async function waitUntilVarIsDefined(r, timeout) {
+    let timeWas = new Date();
+    let result = false;
+    do {
+        result = await isVarDefined(r.variables.session_jwt);
+    } while ((! result) && (new Date() - timeWas < timeout));
```

Comment:
May be subtract or sum poll intervals instead? We don't really need actual time here.

Reply:
Good point, the use of Date() gives a more accurate timeout, but clearly such accuracy is not important for us in this case. Fixed.